lib/rules/require-default-props.js

```diff
@@ -141,6 +141,11 @@ module.exports = {
       switch (node.typeAnnotation.type) {
         case 'GenericTypeAnnotation':
           var annotation = resolveGenericTypeAnnotation(node.typeAnnotation);
+
+          if (annotation && annotation.id) {
+            annotation = findVariableByName(annotation.id.name);
+          }
+
           properties = annotation ? annotation.properties : [];
```

Comment:
You should also update the next line to check for `annotation.properties` existence, or the rule will keep crashing if there is multiple reassignments

```js
properties = annotation && annotation.properties ? annotation.properties : [];
```